backendprocesses.cpp

```diff
@@ -28,51 +32,32 @@ BackendProcesses::BackendProcesses(QByteArray &bytes, std::vector<std::string> &
     this->tstampOffsets.ms = timeDataOffsets.ms;
 }
 
-/*BackendProcesses::~BackendProcesses(){}*/
-
-void BackendProcesses::onNewConnection()
-{
-   QTcpSocket *clientSocket = _server.nextPendingConnection();
-   //connect(clientSocket, SIGNAL(readyRead()), this, SLOT(onReadyRead()));
-   connect(clientSocket, SIGNAL(stateChanged(QAbstractSocket::SocketState)), this, SLOT(onSocketStateChanged(QAbstractSocket::SocketState)));
-
-    _sockets.push_back(clientSocket);
-    emit eng_dash_connection(1);
-    /*for (QTcpSocket* socket : _sockets) {
-        socket->write(QByteArray::fromStdString("From solar car: " + clientSocket->peerAddress().toString().toStdString() + " connected to server !\n"));
-    }*/
+void BackendProcesses::comm_status(bool s) {
+    emit eng_dash_connection(s);
 }
 
-void BackendProcesses::onSocketStateChanged(QAbstractSocket::SocketState socketState)
-{
-    if (socketState == QAbstractSocket::UnconnectedState)
-    {
-        QTcpSocket* sender = static_cast<QTcpSocket*>(QObject::sender());
-        _sockets.removeOne(sender);
-        emit eng_dash_connection(0);
-    }
-}
+void BackendProcesses::startThread() {
+    std::vector<DTI*> obj(1); //create a bunch of DTI instances and add them into this array in order of priority to be sent to telemetrylib
+    long long first_msec = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
 
-/*void BackendProcesses::onReadyRead()
-{
-    QTcpSocket* sender = static_cast<QTcpSocket*>(QObject::sender());
-    QByteArray datas = sender->readAll();
-    for (QTcpSocket* socket : _sockets) {
-        if (socket != sender)
-            socket->write(QByteArray::fromStdString(sender->peerAddress().toString().toStdString() + ": " + datas.toStdString()));
-    }
-}*/
-
-void BackendProcesses::startThread()
-{
-    _server.listen(QHostAddress::AnyIPv4, 4003);
-    connect(&_server, SIGNAL(newConnection()), this, SLOT(onNewConnection()));
+    obj[0]=new SQL(QString::fromStdString(std::to_string(first_msec))); //This sends data to the cloud server
+    //obj[1]=new TCP(QHostAddress::AnyIPv4, 4003); //this sends data thru TCP sockets
```

Comment:
Oops. We still need TCP. Also, line 40 will have to be changed to instantiate 2 objs I believe

Reply:
Accidentally pushed my test code